web-bundle/src/main/java/com/graphhopper/replica/GtfsLinkMapper.java

```diff
@@ -115,89 +114,86 @@ public void setGtfsLinkMappings() {
+
+                int pairCount = pairCountAtomic.incrementAndGet();
+                boolean shouldLog = (
+                        uniqueStopPairs.size() > 10 &&
+                        pairCount % (uniqueStopPairs.size() / 10) == 0
+                );
+                if (shouldLog) {
+                    logger.info("Processed ~" + pairCount + "/" + uniqueStopPairs.size() + " stop pairs so far for feed " + feed.feedId);
+                };
+
+                // Form stop->stop auto routing requests and request a route
+                GHRequest odRequest = new GHRequest(
+                        stop.stop_lat, stop.stop_lon,
+                        nextStop.stop_lat, nextStop.stop_lon
+                );
+                odRequest.setProfile("car");
+                odRequest.setPathDetails(Lists.newArrayList("stable_edge_ids"));
+                GHResponse response = graphHopper.route(odRequest);
+
+                // If stop->stop path couldn't be found by GH, don't store anything
+                if (response.getAll().size() == 0 || response.getAll().get(0).hasErrors()) {
+                    routeNotFoundCountAtomic.incrementAndGet();
+                    return;
                 }
-                processedTripCount++;
-            }
-            logger.info("Done processing GTFS feed " + feed.feedId + "; " + tripIdToStopsInTrip.keySet().size() +
-                    " total trips processed; " + nonUniqueODPairs + "/" + odStopCount
-                    + " O/D stop pairs were non-unique; routes for " + routeNotFoundCount + "/" + odStopCount
-                    + " stop->stop pairs were not found");
-
-            gtfsLinkMappingCsvRows.addAll(getGtfsLinkCsvRowsForFeed(routeIdToTripsInRoute, tripIdToStopPairsInTrip, gtfsLinkMappings));
-        }
+
+                // Parse stable IDs for each edge from response
+                List<PathDetail> responsePathEdgeIdDetails = response.getAll().get(0)
+                        .getPathDetails().get("stable_edge_ids");
+                List<String> pathEdgeIds = responsePathEdgeIdDetails.stream()
+                        .map(pathDetail -> (String) pathDetail.getValue())
+                        .collect(Collectors.toList());
+                // allStableIds.addAll(pathEdgeIds);
+
+                // Merge all path IDs into String to use as value for gtfs link map
+                String pathStableEdgeIdString = pathEdgeIds.stream().collect(Collectors.joining(","));
+                gtfsLinkMappings.put(formatStopIds(stop, nextStop), pathStableEdgeIdString);
+            });
+            logger.info("Done processing GTFS feed " + feed.feedId + "; " + uniqueStopPairs.size() +
+                    " total stop pairs processed; routes for " + routeNotFoundCountAtomic.get() +
+                    " stop->stop pairs were not found");
+
+            return getGtfsLinkCsvRowsForFeed(routeIdToStopPairs, gtfsLinkMappings).stream();
+        }).sorted().collect(Collectors.toList());
```

Comment:
I'm not sure how big these get - not crazy big, right? Do we think sorting is something that could slow things down enough to matter? If not, sounds awesome!

Reply:
(also, micronit: might be good to add a newline after 181 just to visually denote the end of the big lambda block where we create all the CSV rows)

Reply:
NA is 189000 rows, so not bad (now that we fixed the ~33x row dup issue haha).